tests/core/test_typing.py

```diff
@@ -208,31 +208,3 @@ def test_infer_union_operator():
     x = makeDistWithType(Union[List[int], Tuple[int, ...]])
     y = x[2]
     assert underlyingType(y) is int
-
-def test_infer_optional_operator():
```

Comment:
Why get rid of these tests (and rename `test_infer_union_operator`)? Maybe something went wrong with a merge?

Reply:
Got lost in the merge, re-added them and fixed some pruning techniques to be compatible. All tests are passing now.